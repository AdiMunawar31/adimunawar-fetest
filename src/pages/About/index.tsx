import React from "react";
import Footer from "../../components/Footer";
import Navbar from "../../components/Navbar";

const About = () => {
  return (
    <>
      <Navbar />
      <div className='h-full flex justify-center items-center ml-4 sm:ml-20 mr-2'>
        <div className='container m-auto text-gray-600 px-6'>
          <div className='space-y-6 mt-20 sm:mt-40  md:space-y-0 md:flex md:gap-6 lg:items-center lg:gap-12'>
            <div className='md:5/12 lg:w-5/12'>
              <img
                src='https://tailus.io/sources/blocks/left-image/preview/images/startup.png'
                alt='image'
                loading='lazy'
              />
            </div>
            <div className='md:7/12 lg:w-6/12'>
              <h2 className='text-2xl text-gray-900 font-bold md:text-4xl -mt-8'>
                <span className='text-blue-900 mr-2'>Adi Munawar Frontend</span>
                <span className='text-cyan-500'>(DiContact)</span>
              </h2>
              <p className='mt-6 text-gray-600'>
                Using this contact management app has been nothing short of a
                game-changer. As a professional constantly on the move, staying
                in touch with my network used to be a daunting task. However,
                this app has transformed my approach to communication and
                organization. The automated scans for out-of-date information
                are an absolute lifesaver. I no longer need to worry about
                whether I have the most accurate contact details for my network.
                The peace of mind that comes with always having up-to-date
                information is priceless.
              </p>
              <p className='mt-6 text-gray-600'>
                Not only does this app simplify my life, but it also enhances my
                productivity. It's incredibly user-friendly and the intuitive
                interface allows me to effortlessly manage my contacts. No more
                fumbling through endless lists or spreadsheets. It's like having
                a personal assistant that keeps my digital Rolodex in perfect
                order.
              </p>
              <p className='mt-4 text-gray-600'>
                {" "}
                Furthermore, the app's seamless communication capabilities have
                strengthened my professional relationships. I can now connect
                with confidence, knowing that I have the most current
                information at my fingertips.
              </p>
            </div>
          </div>
        </div>
      </div>
      <Footer />
    </>
  );
};

export default About;
