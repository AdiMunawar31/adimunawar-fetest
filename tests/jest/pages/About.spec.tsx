import React from "react";
import renderer from "react-test-renderer";
import About from "@pages/About";
import { MemoryRouter } from "react-router-dom";

describe("About component", () => {
  it("renders the About component", () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={["/not-found"]}>
        <About />
      </MemoryRouter>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
