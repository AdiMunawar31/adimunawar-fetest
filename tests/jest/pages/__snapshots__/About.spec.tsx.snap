// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component renders the About component 1`] = `
[
  <header>
    <nav
      className="fixed z-20 w-full bg-white/90 dark:bg-gray-900/80 backdrop-blur navbar shadow-2xl shadow-gray-600/5 border-b border-gray-100 dark:border-gray-800 peer-checked:navbar-active dark:shadow-none"
    >
      <div
        className="xl:container m-auto px-6 md:px-12 lg:px-6"
      >
        <div
          className="flex flex-wrap items-center justify-between gap-6 md:py-3 md:gap-0 lg:py-5"
        >
          <div
            className="w-full items-center flex justify-between lg:w-auto"
          >
            <a
              aria-label="logo"
              href="/"
              onClick={[Function]}
            >
              <div
                className="flex"
              >
                <img
                  alt="logo"
                  src="/contact.png"
                  width={40}
                />
                <span
                  className="text-lg text-cyan-400 font-bold ml-2 mt-2"
                >
                  DiContact
                </span>
              </div>
            </a>
            <button
              className="peer-checked:hamburger block relative z-20 p-6 -mr-6 cursor-pointer sm:hidden"
              onClick={[Function]}
            >
              <div
                aria-hidden="true"
                className="m-auto h-0.5 w-5 rounded bg-gray-900 dark:bg-gray-300 transition duration-300"
              />
              <div
                aria-hidden="true"
                className="m-auto mt-2 h-0.5 w-5 rounded bg-gray-900 dark:bg-gray-300 transition duration-300"
              />
            </button>
          </div>
          <div
            className="navmenu hidden w-full flex-wrap justify-end items-center mb-16 space-y-8 p-6 border border-gray-100 rounded-3xl shadow-2xl shadow-gray-300/20 bg-white sm:space-y-0 sm:p-0 sm:m-0 sm:flex sm:flex-nowrap sm:bg-transparent sm:w-10/12 sm:shadow-none sm:border-0"
          >
            <div
              className="text-gray-600 sm:pr-4"
            >
              <ul
                className="space-y-6 tracking-wide font-medium text-base sm:text-sm sm:flex sm:space-y-0"
              >
                <li>
                  <a
                    className="block md:px-4 transition hover:text-primary"
                    href="/"
                    onClick={[Function]}
                  >
                    <span
                      aria-label="Home"
                    >
                      Home
                    </span>
                  </a>
                </li>
                <li>
                  <a
                    className="block md:px-4 transition hover:text-primary"
                    href="/about"
                    onClick={[Function]}
                  >
                    <span
                      aria-label="About"
                    >
                      About
                    </span>
                  </a>
                </li>
              </ul>
            </div>
            <div
              className="w-full space-y-2 border-primary/10 flex flex-col ml-1 sm:flex-row lg:space-y-0 md:w-max lg:border-l"
            >
              <a
                aria-label="Dashboard"
                className="relative flex h-9 -ml-2 sm:ml-8 items-center justify-center sm:px-6 before:absolute before:inset-0 before:rounded-full before:bg-blue-900 before:transition before:duration-300 hover:before:scale-105 active:duration-75 active:before:scale-95"
                href="/dashboard"
                onClick={[Function]}
              >
                <span
                  aria-label="Dashboard"
                  className="relative text-sm font-semibold text-white dark:text-gray-900"
                >
                  Dashboard
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </header>,
  <div
    className="h-full flex justify-center items-center ml-4 sm:ml-20 mr-2"
  >
    <div
      className="container m-auto text-gray-600 px-6"
    >
      <div
        className="space-y-6 mt-20 sm:mt-40  md:space-y-0 md:flex md:gap-6 lg:items-center lg:gap-12"
      >
        <div
          className="md:5/12 lg:w-5/12"
        >
          <img
            alt="image"
            loading="lazy"
            src="https://tailus.io/sources/blocks/left-image/preview/images/startup.png"
          />
        </div>
        <div
          className="md:7/12 lg:w-6/12"
        >
          <h2
            className="text-2xl text-gray-900 font-bold md:text-4xl -mt-8"
          >
            <span
              className="text-blue-900 mr-2"
            >
              Adi Munawar Frontend
            </span>
            <span
              className="text-cyan-500"
            >
              (DiContact)
            </span>
          </h2>
          <p
            className="mt-6 text-gray-600"
          >
            Using this contact management app has been nothing short of a game-changer. As a professional constantly on the move, staying in touch with my network used to be a daunting task. However, this app has transformed my approach to communication and organization. The automated scans for out-of-date information are an absolute lifesaver. I no longer need to worry about whether I have the most accurate contact details for my network. The peace of mind that comes with always having up-to-date information is priceless.
          </p>
          <p
            className="mt-6 text-gray-600"
          >
            Not only does this app simplify my life, but it also enhances my productivity. It's incredibly user-friendly and the intuitive interface allows me to effortlessly manage my contacts. No more fumbling through endless lists or spreadsheets. It's like having a personal assistant that keeps my digital Rolodex in perfect order.
          </p>
          <p
            className="mt-4 text-gray-600"
          >
             
            Furthermore, the app's seamless communication capabilities have strengthened my professional relationships. I can now connect with confidence, knowing that I have the most current information at my fingertips.
          </p>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="bg-white dark:bg-gray-900"
  >
    <div
      className="container mx-auto px-6 py-12 mt-20"
    >
      <div
        className="md:-mx-3 md:flex md:items-center md:justify-between"
      >
        <h1
          className="text-3xl font-semibold tracking-tight text-gray-800 dark:text-white md:mx-3 xl:text-4xl"
        >
          Subscribe our newsletter to get update.
        </h1>
        <div
          className="mt-6 shrink-0 md:mx-3 md:mt-0 md:w-auto"
        >
          <a
            className="inline-flex w-full items-center justify-center rounded-lg bg-blue-900 px-4 py-2 text-sm text-white duration-300 hover:bg-gray-700 focus:ring focus:ring-gray-300 focus:ring-opacity-80"
            href="/subsribe-dicontact"
          >
            <span
              className="mx-2"
            >
              Subscribe Now
            </span>
            <svg
              className="mx-2 h-6 w-6"
              fill="none"
              stroke="currentColor"
              strokeWidth="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17.25 8.25L21 12m0 0l-3.75 3.75M21 12H3"
                strokeLinecap="round"
                strokeLinejoin="round"
              />
            </svg>
          </a>
        </div>
      </div>
      <hr
        className="my-6 border-gray-200 dark:border-gray-700 md:my-10"
      />
      <div
        className="flex flex-col items-center justify-between sm:flex-row"
      >
        <a
          className="text-2xl font-bold text-gray-800 transition-colors duration-300 hover:text-gray-700 dark:text-white dark:hover:text-gray-300"
          href="/"
          onClick={[Function]}
        >
          <div
            className="flex"
          >
            <img
              alt="logo"
              src="/contact.png"
              width={40}
            />
            <span
              className="text-lg text-cyan-400 font-bold ml-2 mt-2"
              id="DiContact"
            >
              DiContact
            </span>
          </div>
        </a>
        <p
          className="mt-4 text-sm text-gray-500 dark:text-gray-300 sm:mt-0"
        >
          Â© Copyright 2023. All Rights Reserved.
        </p>
      </div>
    </div>
  </footer>,
]
`;
